src/tests/api-tests.ts

```diff
@@ -10813,6 +10813,58 @@ describe('api tests', () => {
     expect(addressEvents.status).toBe(400);
   });
 
+  test('/microblock/:hash duplicate txs', async () => {
+    const microblock_hash = '0x0fff';
+    const microblock = new TestMicroblockStreamBuilder().addMicroblock({ microblock_hash }).build();
+    // adding a tx in a block with canonical and microblock_canonical true
+    const block1 = new TestBlockBuilder({ block_hash: '0x1234', block_height: 1 })
+      .addTx({
+        microblock_canonical: true,
+        // tx is in the same microblock_hash
+        microblock_hash,
+        canonical: true,
+        block_hash: '0x1234',
+      })
```

Comment:
This tx should be added to the `TestMicroblockStreamBuilder` and then written to the db using `updateMicroblocks`